Kernel/Devices/VMWareSVGADevice.cpp

```diff
@@ -0,0 +1,280 @@
+    PCI::enumerate([&](const PCI::Address& address, PCI::ID id) {
+        if (id == vmware_id) {
+            // Get the base address of the IO addresses
+            m_io_base = PCI::get_BAR0(address) & 0xfffffff0;
+            // Get the base address of the pre-allocated (non system ram) framebuffer
+            m_framebuffer_address_physical = PhysicalAddress(page_base_of(PCI::get_BAR1(address)));
+            m_framebuffer_max_size = PCI::get_BAR_space_size(address, 1);
+            // Get the base address of the fifo and map it into kernel space
+            // TODO: do we ever need to read from it?
+            m_fifo_mapping = MM.allocate_kernel_region(PhysicalAddress(page_base_of(PCI::get_BAR2(address) & 0xfffffff0)), page_round_up(PCI::get_BAR_space_size(address, 2)), "VMWareSVGA Driver FIFO", Region::Access::Read | Region::Access::Write, Region::Cacheable::No);
+            m_fifo_addr = m_fifo_mapping->vaddr();
+            // Get the card interrupt line
+            m_interrupt_line = PCI::get_interrupt_line(address);
+            change_irq_number(m_interrupt_line);
+            dbgln("VMWareSVGA: pci address: {}, io_base: {:x}, framebuffer phys: {:x}", address, m_io_base, m_framebuffer_address_physical.as_ptr());
+            // Clear framebuffer memory
+            OwnPtr<Region> clear_map = MM.allocate_kernel_region(m_framebuffer_address_physical, page_round_up(PCI::get_BAR_space_size(address, 1)), "VMWareSVGA FB Test", Region::Access::Write, Region::Cacheable::No);
+            memset(clear_map->vaddr().as_ptr(), 0x00, PCI::get_BAR_space_size(address, 1));
+            clear_map->unmap();
+        }
+    });
+    VERIFY(m_io_base);
+    VERIFY(m_framebuffer_address_physical.as_ptr());
+    VERIFY(m_fifo_addr.as_ptr());
+
+    // Determine SVGA Version
+    // The way this is done ist by writing the highest version we support (2) into a device support register,
+    // and checking if the value has been changed.
+    // If it hasnt, then we found the highest version that both we and the "card" support.
+    // If it has, decrement by one and repeat.
+    m_device_version = make_version_id(2);
+    set_register(IORegister::ID, m_device_version);
+    while (get_register(IORegister::ID) != m_device_version && m_device_version >= make_version_id(0)) {
+        m_device_version--;
+        set_register(IORegister::ID, m_device_version);
+    }
+
+    VERIFY(m_device_version >= make_version_id(0));
+
+    // We want to read the size of the Fifo memory
+    m_fifo_size = get_register(IORegister::MemSize);
+
+    // The reference driver checks some sizes here, but Qemu actually makes its buffers for the fifo smaller then VMWare does, so they fail. The fifo still works, however, and the numbers are accurate.
+
+    // Check if the card is new enough and get the capabilites register
+    if (m_device_version >= make_version_id(1)) {
+        m_device_capabilites = get_register(IORegister::Capabilities);
+    }
```

Comment:
A few typos (you'll have to rename the member globally)
```suggestion
    // Check if the card is new enough and get the capabilities register
    if (m_device_version >= make_version_id(1)) {
        m_device_capabilities = get_register(IORegister::Capabilities);
    }
```